### ISSUE TITLE
No support for short name argument with no space before value

### ISSUE THREAD

**Issue by Qartar** [jarro2783/cxxopts#158]
`getopt` supports the following syntax [[link](https://www.gnu.org/software/libc/manual/html_node/Example-of-Getopt.html)]:

```
% testopt -cfoo
aflag = 0, bflag = 0, cvalue = foo
```
However `cxxopts` throws `option_requires_argument_exception` for the same construction.

**Comment by jarro2783** [comment]
That could probably be made to work. It conflicts with `-abcd`, but I guess I could check whether `bcd` are arguments and if not treat them as a parameter to `a` instead.

**Comment by kimwalisch** [comment]
My [primesieve](https://github.com/kimwalisch/primesieve#command-line-options) application also supports command-line options without a space between the option and value. So currently I cannot switch to **cxxopts** because this would break backwards compatibility.

**Comment by hzhangxyz** [comment]
hope cxxopts support short args without space since it is really common.

**Comment by yurikhan** [comment]
> That could probably be made to work. It conflicts with `-abcd`, but I guess I could check whether `bcd` are arguments and if not treat them as a parameter to `a` instead.

The usual practice is to treat each letter as a short option until you find an option that takes a parameter. At that point, take the remainder of this argument as the parameter value.

So, if `-a` is a short option without parameter, `-b` is a short option taking a string parameter, and `-c` is a short option, then `-abcd` means `-a -b=cd`.

**Comment by eyalroz** [comment]
I also need this, to implement compiler like `-DMACRO=expansion` and `-DMACRO` parsing, as well as `-fcompilation-option`.

> That could probably be made to work. It conflicts with -abcd, but I guess I could check whether bcd are arguments and if not treat them as a parameter to a instead.

I would suggest explicitly marking options which take a no-space argument, so that if `-a` is marked like that, and you get `-abcd`, you _won't_ try to match `b`, `c` or `d` to switches.

**Comment by artpaul** [comment]
> > That could probably be made to work. It conflicts with `-abcd`, but I guess I could check whether `bcd` are arguments and if not treat them as a parameter to `a` instead.
> 
> The usual practice is to treat each letter as a short option until you find an option that takes a parameter. At that point, take the remainder of this argument as the parameter value.
> 
> So, if `-a` is a short option without parameter, `-b` is a short option taking a string parameter, and `-c` is a short option, then `-abcd` means `-a -b=cd`.

Agree with that. Implementing this logic is quite easy and works fine for me:
https://github.com/artpaul/cxxopts/commit/8c57e19d50b4d879943d2396b1a3a52d1f067c67

**Comment by jarro2783** [comment]
That sounds pretty reasonable. Are you able to make a PR out of that?

**Comment by eyalroz** [comment]
@jarro2783 : I'll give it a shot and report back here.

**Comment by eyalroz** [comment]
@artpaul : You're also welcome to play with the feature branch and see if it works or whether there are any bugs.

**Comment by eyalroz** [comment]
@jarro2783 : Your v3.0.0 seems to suggest this issue is already fixed ("Support single letter options with argument attached.") - perhaps you meant something else?

**Comment by jarro2783** [comment]
Looking at the example program, it looks like this works:
```
src/example -aifile
Saw option ‘a’ 1 times 
Input = file
```
But I'm not sure how you would implement something like `-DMACRO=value`

**Comment by eyalroz** [comment]
> But I'm not sure how you would implement something like -DMACRO=value

Oh, I was thinking that this would be up to the application. i.e. the switch is `-D`, and then the value is either `MACRO` or `MACRO=value` together. The "inner value" shouldn't be recognized by cxxopts; I don't think it makes sense for cxxopts itself to add a completely new semantic layer of named "sub-options" with values.

**Comment by jarro2783** [comment]
Ah I understand. I'm not sure if it will parse with the `=` at the moment, that's all.
But yes it sounds like this bug is fixed in 3.0.

**Comment by eyalroz** [comment]
So, this is not fully resolved, as my new issues suggests.
